Mathlib/Combinatorics/SimpleGraph/ConcreteColorings.lean

```diff
@@ -0,0 +1,77 @@
+/-
+Copyright (c) 2023 Iván Renison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Iván Renison
+-/
+import Mathlib.Combinatorics.SimpleGraph.Coloring
+import Mathlib.Combinatorics.SimpleGraph.Hasse
+import Mathlib.Data.ZMod.Basic
+
+/-!
+# Concrete colorings of common graphs
+
+This file defines colorings for some common graphs
+
+## Main declarations
+
+* `SimpleGraph.pathGraph.bicoloring`: Bicoloring of a path graph
+-/
+
+theorem suc_mod2_neq (m : ℕ) : (m % 2 == 0) ≠ ((m + 1) % 2 == 0) := by
+  intro h
+  rw [Bool.eq_iff_eq_true_iff, beq_iff_eq] at h
+  simp only [beq_iff_eq, ←Nat.even_iff, Nat.even_add_one] at h
+  exact not_iff_self h.symm
+
+/-- Bicoloring of a path graph -/
+def SimpleGraph.pathGraph.bicoloring (n : ℕ) :
+    SimpleGraph.Coloring (pathGraph n) Bool :=
+  Coloring.mk (fun u ↦ u.val % 2 == 0)
+    (by
+      intro u v h
+      simp
+      have : u ⋖ v ∨ v ⋖ u := by simp_all [pathGraph]
+      have : u.val ⋖ v.val ∨ v.val ⋖ u.val := by simp_all [Fin.coe_covby_iff]
+      have : u.val + 1 = v.val ∨ v.val + 1 = u.val := by simp_all [Nat.covby_iff_succ_eq]
+      match this with
+        | Or.inl h' =>
+          rw [←h']
+          exact suc_mod2_neq u
+        | Or.inr h' =>
+          rw [←h']
+          exact (suc_mod2_neq v).symm)
+
+/-- Convert a coloring to bool to a coloring to Fin 2 -/
+def SimpleGraph.Coloring.BoolToFin2 {α} {G : SimpleGraph α} (c : SimpleGraph.Coloring G Bool) :
+    SimpleGraph.Coloring G (Fin 2) :=
+  (SimpleGraph.recolorOfEquiv G (finTwoEquiv)).invFun c
```

Comment:
This definition ends up being unnecessary. Lean can unfold things and see that `Fintype.card Bool = 2` without needing it to be `Fintype.card (Fin 2) = 2` (I guessing that's why you introduced this).